dynawo/sources/Modeler/DataInterface/PowSyblIIDM/test/TestCriteria.cpp

```diff
@@ -2059,6 +2161,33 @@ TEST(DataInterfaceIIDMTest, testLoadCriteriaDataIIDMSum) {
   ASSERT_FALSE(data->checkCriteria(0, true));
 }
 
+TEST(DataInterfaceIIDMTest, testDontTestFictitiousLoadsInCriteria) {
+  boost::shared_ptr<CriteriaParams> criteriaParams = CriteriaParamsFactory::newCriteriaParams();
+  criteriaParams->setType(CriteriaParams::LOCAL_VALUE);
+  criteriaParams->setScope(CriteriaParams::DYNAMIC);
+  criteriaParams->setPMax(90);
+
+  shared_ptr<DataInterface> data1 = createDataItfFromNetworkCriteria(createBusBreakerNetworkWithOneFictitiousLoadAmongTwo(180, 190, 100, 100));
+  exportStates(data1);
+  std::vector<boost::shared_ptr<LoadInterface> > loads1 = data1->getNetwork()->getVoltageLevels()[0]->getLoads();
+  LoadCriteria criteria1(criteriaParams);
+  for (size_t i = 0; i < loads1.size(); ++i)
+    criteria1.addLoad(loads1[i]);
+  criteria1.checkCriteria(0, false);
+  const std::vector<std::pair<double, std::string> > failingCriteria1 = criteria1.getFailingCriteria();
+  ASSERT_EQ(failingCriteria1.size(), 1);
+
+  shared_ptr<DataInterface> data2 = createDataItfFromNetworkCriteria(createBusBreakerNetworkWithLoads(180, 190, 100, 100));
```

Comment:
Isn't this test already done somewhere else?

Reply:
Yes, maybe it's useless to check that all loads are checked if all loads are fictitious.